crates/biome_diagnostics/src/location.rs

```diff
@@ -133,7 +133,7 @@ impl LineIndex {
         // SAFETY: Transmuting `&[TextSize]` to `&LineIndex` is safe since
         // `LineIndex` is a `repr(transparent)` struct containing a `[TextSize]`
         // and thus has the same memory layout
-        unsafe { std::mem::transmute(slice) }
+        unsafe { &*((slice as *const [TextSize]) as *const LineIndex) }
```

Comment:
I'd like to have this reverted, it's not very readable, while the transmute is more immediate (Devs can read the documentation)